vllm/config/compilation.py

```diff
@@ -675,6 +675,9 @@ def init_backend(self, vllm_config: "VllmConfig") -> str | Callable:
         ]:
             if self.backend in torch_backends:
                 return self.backend
+            if self.backend == "mirage":
```

Comment:
Let's call this `mirage_hardcoded`/`mirage_direct`/`mirage_byname`, and call the proper torch.compile Mirage backend `mirage`.

Also, we should add 3 things to `VllmConfig.__post_init__`:
- automatically set attention backend to mirage if the compile backend is mirage
- raise a ValueError if attention backend is mirage but compile backend isn't